proxy/http.go

```diff
@@ -8,6 +8,16 @@ import (
 
 func newHTTPProxy(t *url.URL, tr http.RoundTripper) http.Handler {
 	rp := httputil.NewSingleHostReverseProxy(t)
+
+	originalDirector := rp.Director
```

Comment:
Can you explain better in the comment why this is necessary and which problem it solves? Also, `s/originalDirector/d/`